TAO/TAO_IDL/driver/drv_preproc.cpp

```diff
@@ -147,14 +151,60 @@ DRV_cpp_putarg (const char *str)
           buf[0] = ACE_TEXT ('"');
           ACE_OS::strcpy (buf + 1, ACE_TEXT_CHAR_TO_TCHAR (str));
           ACE_OS::strcat (buf, ACE_TEXT ("\""));
-          DRV_arglist[DRV_argcount++] = buf;
+          arg_to_add = buf;
+        }
+      else
+        {
+          idl_global->err()->misc_error ("DRV_cpp_putarg failed to allocate buffer!");
+          throw Bailout ();
         }
     }
-  else
+  else if (str)
     {
-      DRV_arglist[DRV_argcount++] =
-        ACE::strnew (ACE_TEXT_CHAR_TO_TCHAR (str));
+#ifdef ACE_WIN32
+      // Escape Doublequotes on Windows
+      size_t quote_count = 0;
+      for (const char* quote = ACE_OS::strchr (str, '"'); quote; quote = ACE_OS::strchr (quote, '"'))
+        {
+          ++quote_count;
+          ++quote;
+        }
+      if (quote_count)
+        {
+          ACE_TCHAR *buf = nullptr;
+          ACE_NEW_NORETURN (buf, ACE_TCHAR[ACE_OS::strlen (str) + quote_count + 1]);
+          if (buf)
+            {
+              ACE_TCHAR *to = buf;
+              for (const char* from = str; *from; ++from)
+                {
+                  if (*from == '"')
+                    {
+                      *to = ACE_TEXT ('\\');
+                      ++to;
+                    }
+                  *to = *from;
```

Comment:
These are (potentially) different types, char and tchar.  Does this assignment always work?  At least add a comment?

Reply:
Assuming `char` is ASCII and `wchar` encoding is ASCII-based it should just should widen the value like an int cast, but yeah that's probably assuming too much. Probably needs to convert the whole str first.

Reply:
Actually I realized that's insane to work with `ACE_TCHAR` when I didn't have to and did the opposite of converting it first by converting it at the very end.